gdal_subsetter/transform.py

```diff
@@ -244,30 +241,44 @@ def process_geotiff(
             result = self.add_to_result(filelist, output_dir)
             layernames.append(layer_id)
         else:
-            # all possible variables (band names) in the input_filenanme
-            variables = self.get_variables(input_filename)
+            # Variables were specified, so only process requested variables
+            # First get all bands and standard names in the input GeoTIFF file
+            geotiff_variables = get_geotiff_variables(input_filename)
+
             # source.process('variables') is user's requested variables (bands)
-            for variable in source.process("variables"):
-                band = None
-
-                for key, value in variables.items():
-                    if "band" in variable.name.lower():
-                        if variable.name in key:
-                            band = int(key.split("Band")[-1])
-                            break
-                    else:
-                        if variable.name.lower() in value.lower():
-                            band = int(key.split("Band")[-1])
-                            break
+            for requested_variable in source.process("variables"):
+                # First check the standard_name of all bands:
+                band = next(
+                    (
+                        band_name
+                        for band_name, standard_name in geotiff_variables.items()
+                        if requested_variable.name.lower() in standard_name.lower()
+                    ),
+                    None,
+                )
+
+                if band is None:
+                    # No standard name matched, now try raw band names, i.e. BandN
+                    band = next(
+                        (
+                            band_name
+                            for band_name in geotiff_variables
+                            if requested_variable.name.lower() in band_name.lower()
+                        ),
+                        None,
+                    )
 
-                if band:
+                if band is not None:
+                    band_number = int(band.split("Band")[-1])
                     filename = input_filename
-                    layer_id = basename + "__" + variable.name
+                    layer_id = basename + "__" + requested_variable.name
                     layer_id, filename, output_dir = self.perform_transforms(
-                        layer_id, filename, output_dir, band
+                        layer_id, filename, output_dir, band_number
                     )
                     layernames.append(layer_id)
                     filelist.append(filename)
+                else:
+                    raise MissingVariableError(requested_variable.name)
```

Comment:
This is a behaviour change: previously, if a requested variable could not be matched to anything in the GeoTIFF, the service would just carry on silently to the next requested variable. I think that goes against the general theme of: if a user explicitly asks for something, and a service can't do it, then we should fail and say so.

Reply:
Also, generally, I'm lacking test coverage of methods on this class. My plan is:

* Pull unnecessary things from the class, to reduce overall size. Add unit tests to things pulled out.
* Rely on the regression tests for now on the larger behaviour.
* Once the adapter is cut down to size, start adding more tests in the repository for methods still there.

If that just sounds lazy, though, let me know and I can look into adding some tests for this method.